src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -723,4 +767,365 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
+				// }
+				// if (isContext) {
+				//     cellClasses += ' context';
+				// }
+				// if (isHighlighted === true) {
+				//     cellClasses += ' highlighted';
+				// }
+				// if (isShadowed && !isHighlighted && !isOverridden) {
+				//     cellClasses += ' blurred';
+				// }
+				return '<span title="' + valueToDisplay + '" class="' + cellClasses + '">' + valueToDisplay + '</span>';
+			}
+			return undefined;
+		};
+	}
+
+	handleChanges(changes: { [propName: string]: any }): void {
+		let columnDefinitionChanges = changes['columnDefinitions'];
+		let activeCell = this._tables[0] ? this._tables[0].grid.getActiveCell() : undefined;
+		let hasGridStructureChanges = false;
+		let wasEditing = this._tables[0] ? !!this._tables[0].grid.getCellEditor() : false;
+
+		if (columnDefinitionChanges && !equals(columnDefinitionChanges.previousValue, columnDefinitionChanges.currentValue)) {
+			if (!this._tables[0]) {
+				this.createNewTable();
+			} else {
+				this._tables[0].grid.resetActiveCell();
+				this._tables[0].grid.setColumns(this.dataSet.columnDefinitions);
+			}
+			hasGridStructureChanges = true;
+
+			if (!columnDefinitionChanges.currentValue || columnDefinitionChanges.currentValue.length === 0) {
+				activeCell = undefined;
+			}
+			if (activeCell) {
+				let columnThatContainedActiveCell = columnDefinitionChanges.previousValue[Math.max(activeCell.cell - 1, 0)];
+				let newActiveColumnIndex = columnThatContainedActiveCell
+					? columnDefinitionChanges.currentValue.findIndex(c => c.id === columnThatContainedActiveCell.id)
+					: -1;
+				activeCell.cell = newActiveColumnIndex !== -1 ? newActiveColumnIndex + 1 : 0;
+			}
+		}
+
+		if (changes['dataRows']
+			|| (changes['highlightedCells'] && !equals(changes['highlightedCells'].currentValue, changes['highlightedCells'].previousValue))
+			|| (changes['blurredColumns'] && !equals(changes['blurredColumns'].currentValue, changes['blurredColumns'].previousValue))
+			|| (changes['columnsLoading'] && !equals(changes['columnsLoading'].currentValue, changes['columnsLoading'].previousValue))) {
+			this.setCallbackOnDataRowsChanged();
+			this._tables[0].grid.updateRowCount();
+			this._tables[0].grid.setColumns(this._tables[0].grid.getColumns());
+			this._tables[0].grid.invalidateAllRows();
+			this._tables[0].grid.render();
+			hasGridStructureChanges = true;
+		}
+
+		if (hasGridStructureChanges) {
+			if (activeCell) {
+				this._tables[0].grid.setActiveCell(activeCell.row, activeCell.cell);
+			} else {
+				this._tables[0].grid.resetActiveCell();
+			}
+		}
+
+		if (wasEditing && hasGridStructureChanges) {
+			console.log('need to handle editing of cells');
+			//this._tables[0].grid.editActiveCell();
+		}
+	}
+
+	private setCallbackOnDataRowsChanged(): void {
+		if (this.dataSet.dataRows) {
+			// We must wait until we get the first set of dataRows before we enable editing or slickgrid will complain
+			console.log('need to check if row is editable or not');
```

Comment:
LogService?  I won't go through each `console.log` but I assume they all should generally be using the LogService instead?